src/perception/test/performance_test.cpp

```diff
@@ -0,0 +1,160 @@
+                     long long RANSAC_time,
+                     long long ground_removed_size,
+                     long long DBSCAN_time,
+                     long long ConeDif_time,
+                     long long Total_time,
+                     long long generatedClusters,
+                     long long blues,
+                     long long yellows,
+                     long long undefineds,
+                     long long conversion_duration) {
+        bool fileExists = std::filesystem::exists(output_statistics_path_file);
+
+        std::ofstream csv_file(output_statistics_path_file, std::ios::app);
+        if (!fileExists) {
+            csv_file << "Initial Number of Points,Conversion Duration,"
+                        << "RANSAC Epsilon,RANSAC Number Iterations,"
+                        << "Number of Points after RANSAC,RANSAC Duration(ms),"
+                        << "DBSCAN Distance Threshold,DBSCAN Neighbours Threshold,"
+                        << "Number of Generated Clusters,"
+                        << "DBSCAN Duration(ms),"
+                        << "Number of Blue Cones,Number of Yellow Cones,"
+                        << "Number of Undefined Cones,ConeDifferentiation Duration(ms),"
+                        << "Total Duration(ms)\n";
+        }
+
+        csv_file << std::fixed << std::setprecision(6);
+        csv_file << initial_n_points << ","
+            << std::fixed << std::setprecision(3) << conversion_duration << ","
+            << RANSAC_eps << ","
+            << RANSAC_Iter << ","
+            << ground_removed_size << ","
+            << std::fixed << std::setprecision(3) << RANSAC_time << ","
+            << DBSCAN_dist_threshold << ","
+            << DBSCAN_neighbours_threshold << ","
+            << generatedClusters << ","
+            << std::fixed << std::setprecision(3) << DBSCAN_time << ","
+            << blues << ","
+            << yellows << ","
+            << undefineds << ","
+            << std::fixed << std::setprecision(3) << ConeDif_time << ","
+            << std::fixed << std::setprecision(3) << Total_time << std::endl;
+
+        csv_file.close();
+    }
+
+    std::string output_statistics_path_file = "../../performance/exec_time/perception_" +
+                    getCurrentDateTimeAsString() + ".csv";
+    std::string inputFilesPaths = "../../src/perception/test/point_clouds/";
+
+    GroundRemoval* ground_removal;
+    Clustering* clustering;
+    ConeDifferentiation* cone_differentiator;
+    pcl::PointCloud<pcl::PointXYZI>::Ptr pcl_cloud;
+
+    float RANSAC_eps = 0.1;
+    int RANSAC_Iter = 20;
+
+    float DBSCAN_dist_threshold = 0.1;
+    int DBSCAN_neighbours_threshold = 1;
+};
+
+
+
+TEST_F(PerceptionPerformanceTest, TestPerformance) {
+    for (int it = 0; it <= 7; it++) {
+        std::string file_name =  inputFilesPaths + "PointCloud" + std::to_string(it) + ".pcd";
+
+        pcl::io::loadPCDFile<pcl::PointXYZI>(file_name, *pcl_cloud);
```

Comment:
Test format is overall good but I think it would be good to better document this. Separate into steps clearly would help here